sklearn/metrics/cluster/tests/test_common.py

```diff
@@ -0,0 +1,170 @@
+        assert_not_equal(metric(y1, y2), metric(y2, y1),
+                         msg="%s is symmetric" % name)
+
+
+def test_normalized_output():
+    upper_bound_1 = [0, 0, 0, 1, 1, 1]
+    upper_bound_2 = [0, 0, 0, 1, 1, 1]
+    for name in NORMALIZED_METRICS:
+        metric = SUPERVISED_METRICS[name]
+        assert_greater(metric([0, 0, 0, 1, 1], [0, 0, 0, 1, 2]), 0.0)
+        assert_greater(metric([0, 0, 1, 1, 2], [0, 0, 1, 1, 1]), 0.0)
+        assert_less(metric([0, 0, 0, 1, 2], [0, 1, 1, 1, 1]), 1.0)
+        assert_less(metric([0, 0, 0, 1, 2], [0, 1, 1, 1, 1]), 1.0)
+        assert_equal(metric(upper_bound_1, upper_bound_2), 1.0,
+                     msg="%s has upper_bound greater than 1" % name)
+
+    lower_bound_1 = [0, 0, 0, 0, 0, 0]
+    lower_bound_2 = [0, 1, 2, 3, 4, 5]
+    for name in NORMALIZED_METRICS:
+        metric = SUPERVISED_METRICS[name]
+        assert_equal((metric(lower_bound_1, lower_bound_2) and
+                      metric(lower_bound_2, lower_bound_1)), 0.0,
+                     msg="%s has lower_bound less than 0.0" % name)
+
+
+# All clustering metrics do not change score due to permutations of labels
+# that is when 0 and 1 exchchanged.
+def test_permute_labels():
+    y_label = np.array([0, 0, 0, 1, 1, 0, 1])
+    y_pred = np.array([1, 0, 1, 0, 1, 1, 0])
+    for name in SUPERVISED_METRICS:
+        metric = SUPERVISED_METRICS[name]
+        score_1 = metric(y_pred, y_label)
+        assert_equal(score_1, metric(1 - y_pred, y_label),
+                     msg="%s failed labels permutation" % name)
+        assert_equal(score_1, metric(1 - y_pred, 1 - y_label),
+                     msg="%s failed labels permutation" % name)
+        assert_almost_equal(score_1, metric(y_pred, 1 - y_label),
+                            err_msg="%s failed labels permutation" % name)
+
+    for name in UNSUPERVISED_METRICS:
+        metric = UNSUPERVISED_METRICS[name]
+        X = np.random.randint(10, size=(7, 10))
+        score_1 = metric(X, y_pred)
+        assert_almost_equal(score_1, metric(X, 1 - y_pred),
+                            err_msg="%s failed labels permutation" % name)
+
+
+# For ALL clustering metrics Input parameters can be both
+# in the form of arrays lists , positive , negetive or string
+def test_format_invariance():
+    labels = [0, 0, 0, 0, 1, 1, 1, 1]
+    y_pred = [0, 1, 2, 3, 4, 5, 6, 7]
+
+    def generate_formats(y):
+        y = np.array(y)
+        y = np.array(y)
+        yield y, 'array of ints'
+        yield y.tolist(), 'list of ints'
+        yield [str(x) for x in y.tolist()], 'list of strs'
+        yield y - 1, 'including negative ints'
+        yield y + 1, 'strictly positive ints'
+
+    for name in SUPERVISED_METRICS:
+        metric = SUPERVISED_METRICS[name]
+        score_1 = metric(labels, y_pred)
+        mygenerator_1 = generate_formats(labels)
+        mygenerator_2 = generate_formats(y_pred)
+        for i, j in zip(mygenerator_1, mygenerator_2):

```

Comment:
can do `for (y_true_fmt, fmt_name), (y_pred_fmt, _) in zip(y_true_gen, y_pred_gen)`